protocol/codec_test.go

```diff
@@ -199,3 +199,29 @@ func TestEncodeJSON(t *testing.T) {
 	require.True(t, reflect.DeepEqual(v, nsv))
 	require.True(t, reflect.DeepEqual(v, sv))
 }
+
+func TestRandomizeObjectWithPtrField(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	type testObjA struct {
+		U64 uint64
+	}
+	type testObjB struct {
+		U16  uint16
+		ObjA *testObjA
+	}
+
+	// run a few and fail if all ints are zero
+	sawNonZeroInt := false
+	for i := 0; i < 10; i++ {
+		obj, err := RandomizeObject(&testObjB{})
+		require.NoError(t, err)
+		objB, ok := obj.(*testObjB)
+		require.True(t, ok)
+		require.NotNil(t, objB.ObjA)
+		if objB.U16 != 0 || objB.ObjA.U64 != 0 {
```

Comment:
This doesn't actually test that U16 and U64 both work. It tests that at least one works.

I'm not terribly worried about this, but if the intent was to independently test that the randomizer works on u64 and u16 types, we can't || the test together.

Reply:
Updated